lib/checkcondition.cpp

```diff
@@ -469,7 +469,7 @@ void CheckCondition::duplicateCondition()
                 }
             }
             if (tok3->varId() > 0 &&
-                isVariableChanged(scope.classDef->next(), cond2, tok3->valueType()->pointer, tok3->varId(), false, mSettings, mTokenizer->isCPP())) {
+                isVariableChanged(scope.classDef->next(), cond2, tok3->valueType() ? tok3->valueType()->pointer : 0, tok3->varId(), false, mSettings, mTokenizer->isCPP())) {
```

Comment:
Why not use `astIsPointer`?

Reply:
`tok3->valueType()->pointer` is an integer (for instance 2 for a pointer to a pointer), I believe isVariableChanged uses this (and not a boolean).

Reply:
I see.